### ISSUE TITLE
Repetitve variable. 

### ISSUE THREAD

**Issue by SiNa-Dehbari** [lark-parser/lark#795]
Hello Everybody.

This is my first parsering project and  I am new in Lark. I am translating a new format language (DSL) and have two questions as below:

1- in the code there are some variables which are combination of string and integer like `vehicle1` or `path2`.  Is there any command for reading them? WORD gets just  string.  ESCAPED_STRING works, but to use it, I should put "" sides of variables at the main code. (preferably I don't want to touch code)

2- in my file, one variable, for example `vehicle1` is defined at first lines (`"Vehicle1": car with:`). in grammar I write:
`car_name:    ESCAPED_STRING ":" "car" "with" ":"`
and Lark gets  the name of this variable.
After that this name is repeated several times in different structure 
`"Vehicle1".drive(path1) with:`

how can I tell to Lark that "vehicle1" is the same variable and get it from the above?
if I define it as a name once again, I get this error :
`GrammarError: Rule 'car_name' defined more than once`

I hope I have explained my problems good.
Thank you in advance.

**Comment by MegaIng** [comment]
1- `CNAME` is probably what you want.

2- You don't tell that lark, you do that in your own steps, e..g with a Transformer or, more likely, an Interpreter. The parser can only do syntax, not semantics.

**Comment by SiNa-Dehbari** [comment]
Thank you @MegaIng . your answers were a great help for me.
But still I am not sure about my next step. I even don't know what do I need!
As I said, I am working with new language format and I want to form it in a common format. a part of my language is a below:

`scenario CCRb:
	V1: car with:
        keep(it.category==car)
        keep(it.color==red)

	path1: path
	
	do serial ():
		initiation: parallel(duration: [1..3]s)
			V1.drive(path1) with:
				speed(speed:50 kph)
				`
I have developed first part (parser) and got below tree.
![photo](https://user-images.githubusercontent.com/71755293/101346945-9d8ef980-3889-11eb-8392-87594f79be9a.png)


the final out put should be something like:

`state_of_Vehicle 1: (path=path1, position =0, speed=50, acceleration=0)
`
after your responding, I looked through the PDF and read about visitor, transformer and interpreter. but they were ambiguous for me (a Mechanical engineer!)

Thank you in advance
